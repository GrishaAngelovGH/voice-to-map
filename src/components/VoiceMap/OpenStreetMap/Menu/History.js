import React from 'react'
import { useSelector } from 'react-redux'
import { Fragment } from 'react'

import Dialog from '@mui/material/Dialog'
import AppBar from '@mui/material/AppBar'
import Toolbar from '@mui/material/Toolbar'
import IconButton from '@mui/material/IconButton'
import Typography from '@mui/material/Typography'
import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemText from '@mui/material/ListItemText'
import Divider from '@mui/material/Divider'
import Slide from '@mui/material/Slide'

import RoomIcon from '@mui/icons-material/Room'
import CloseIcon from '@mui/icons-material/Close'

const Transition = React.forwardRef(function Transition(props, ref) {
    return <Slide direction="up" ref={ref} {...props} />
})

const History = ({ onClose }) => {
    const history = useSelector(state => state.locations.history)

    return (
        <Dialog fullScreen={true} open={true} TransitionComponent={Transition} data-testid="history">
            <AppBar sx={{ position: 'sticky' }}>
                <Toolbar>
                    <IconButton
                        edge="start"
                        color="inherit"
                        onClick={onClose}
                        aria-label="close"
                    >
                        <CloseIcon />
                    </IconButton>
                    <Typography sx={{ ml: 2, flex: 1 }} variant="h6" component="div">
                        History
                    </Typography>
                </Toolbar>
            </AppBar>
            <List>
                {
                    history.map((v, i) => (
                        <Fragment key={v.place_id}>
                            <ListItem button>
                                <ListItemText primary={<span>{`#${i + 1}`} <RoomIcon /> {v.display_name}</span>} />
                            </ListItem>
                            <Divider />
                        </Fragment>
                    ))
                }
            </List>
        </Dialog>
    )
}

export default History