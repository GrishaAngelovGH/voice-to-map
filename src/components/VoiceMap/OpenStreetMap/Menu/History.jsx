import React from 'react'
import { useSelector } from 'react-redux'
import { Fragment } from 'react'

import Dialog from '@mui/material/Dialog'
import AppBar from '@mui/material/AppBar'
import Toolbar from '@mui/material/Toolbar'
import IconButton from '@mui/material/IconButton'
import Typography from '@mui/material/Typography'
import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemText from '@mui/material/ListItemText'
import Divider from '@mui/material/Divider'

import RoomIcon from '@mui/icons-material/Room'
import CloseIcon from '@mui/icons-material/Close'

const History = ({ onClose }) => {
	const history = useSelector(state => state.locations.history)

	return (
		<Dialog
			fullScreen={true}
			open={true}
		>
			<AppBar sx={{ position: 'sticky' }}>
				<Toolbar>
					<IconButton
						edge='start'
						color='inherit'
						onClick={onClose}
						aria-label='close'
					>
						<CloseIcon />
					</IconButton>
					<Typography sx={{ ml: 2, flex: 1 }} variant='h6' component='div'>
						History
					</Typography>
				</Toolbar>
			</AppBar>
			<List>
				{
					history.map((v, i) => (
						<Fragment key={v.place_id}>
							<ListItem>
								<ListItemText primary={<span>{`#${i + 1}`} <RoomIcon /> {v.display_name}</span>} />
							</ListItem>
							<Divider />
						</Fragment>
					))
				}
			</List>
		</Dialog>
	)
}

export default History