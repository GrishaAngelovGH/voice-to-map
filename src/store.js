import { configureStore } from '@reduxjs/toolkit'
import locationReducer from '../src/slices/locationSlice'

const store = configureStore({
    reducer: {
        locations: locationReducer
    }
})

export default store