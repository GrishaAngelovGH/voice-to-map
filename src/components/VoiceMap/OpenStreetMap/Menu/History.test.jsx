import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { setupStore } from 'store'

import History from './History'

const store = setupStore({
	locations: {
		lastLocation: '',
		history: [
			{
				'place_id': '245424010',
				'licence': 'https://locationiq.com/attribution',
				'osm_type': 'way',
				'osm_id': '744663875',
				'boundingbox': [
					'51.501678',
					'51.5017063',
					'-0.1618417',
					'-0.1614632'
				],
				'lat': '51.501691',
				'lon': '-0.1616784',
				'display_name': 'Knightsbridge, Knightsbridge, Westminster, Greater London, England, SW1X 7QS, United Kingdom',
				'class': 'highway',
				'type': 'primary',
				'importance': 0.32
			}
		]
	}
})

test('should render History component', () => {
	const view = render(
		<Provider store={store}>
			<History />
		</Provider>
	)

	expect(view).toMatchSnapshot()
})