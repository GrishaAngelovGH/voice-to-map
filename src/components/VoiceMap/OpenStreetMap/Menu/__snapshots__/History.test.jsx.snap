// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render History component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-1424xw8-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-1cennmq-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-19do60a-MuiDialog-container"
        role="presentation"
        style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          aria-modal="true"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen css-l09ud3-MuiPaper-root-MuiDialog-paper"
          role="dialog"
          style="--Paper-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);"
        >
          <header
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-134gap9-MuiPaper-root-MuiAppBar-root"
            style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12); padding-right: 1024px;"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
            >
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1fytndg-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </button>
              <div
                class="MuiTypography-root MuiTypography-h6 css-ybxm6-MuiTypography-root"
              >
                History
              </div>
            </div>
          </header>
          <ul
            class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-z2qibm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    #1
                     
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RoomIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
                      />
                    </svg>
                     
                    Knightsbridge, Knightsbridge, Westminster, Greater London, England, SW1X 7QS, United Kingdom
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
          </ul>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
